### ISSUE TITLE
NOSCRIPT ReplyError

### ISSUE THREAD

**Issue by TheGame2500** [SGrondin/bottleneck#50]
I keep getting this error with no apparent pattern.

```
ReplyError: NOSCRIPT No matching script. Please use EVAL.
    at parseError (/Users/gabrielbira/workspace/littledata/queue/node_modules/redis-parser/lib/parser.js:193:12)
    at parseType (/Users/gabrielbira/workspace/littledata/queue/node_modules/redis-parser/lib/parser.js:303:14)
  command: 'EVALSHA',
  args: 
   [ undefined,
     3,
     'b_settings',
     'b_running',
     'b_executing',
     '0',
     '1',
     '1518520777288' ],
  code: 'NOSCRIPT'
}
```

**Comment by SGrondin** [comment]
I think I've figured it out.

That error happens because the limiter is trying to interact with the Redis server before Bottleneck is done setting up its Lua scripts.

The limiter emits the `ready` event when it is ready to accept requests but there is currently no safeguard in place to prevent people from submitting requests to a limiter still starting up.

Can you confirm that you don't currently wait for the `ready` event?

**Comment by SGrondin** [comment]
I've just released `v2.1.0` which returns a nice error explaining that the limiter isn't done connecting to Redis yet, and how to know when it's ready to receive requests. You will not be seeing this gnarly error again.

Thanks for your bug report, here's the fix: https://github.com/SGrondin/bottleneck/commit/0117b152b57fd78690a47e606f0ee788b1e32f6b
